File: python/3127. Make a Square with the Same Color.py
class Solution:
    def canMakeSquare(self, grid: List[List[str]]) -> bool:
        for i, j in [[0,0],[0,1],[1,0],[1,1]]:
            count = 0
            for x, y in [[0,0],[0,1],[1,0],[1,1]]:
                count += 1 if grid[i+x][j+y] == 'B' else 0
            if count != 2:
                return True
        return False

File: java/3127. Make a Square with the Same Color.java
class Solution {
    public boolean canMakeSquare(char[][] grid) {
        int[][] directions = new int[][]{{0, 0}, {0, 1}, {1, 0}, {1, 1}};
        for (int[] d : directions) {
            int i = d[0], j = d[1];
            int count = 0;
            for (int[] d1 : directions) {
                int x = d1[0], y = d1[1];
                if (grid[i+x][j+y] == 'B') count++;
            }
            if (count != 2) return true;
        }
        return false;
    }
}